src/transformers/pipelines/object_detection.py

```diff
@@ -0,0 +1,170 @@
+        elif isinstance(image, Image.Image):
+            return image
+
+        raise ValueError(
+            "Incorrect format used for image. Should be an url linking to an image, a local path, or a PIL image."
+        )
+
+    def __call__(
+        self,
+        images: Union[str, List[str], "Image", List["Image"]],
+        threshold: Optional[float] = 0.9,
+    ) -> Union[Predictions, List[Prediction]]:
+        """
+        Detect objects (bounding boxes & classes) in the image(s) passed as inputs.
+
+        Args:
+            images (:obj:`str`, :obj:`List[str]`, :obj:`PIL.Image` or :obj:`List[PIL.Image]`):
+                The pipeline handles three types of images:
+
+                - A string containing a http link pointing to an image
+                - A string containing a local path to an image
+                - An image loaded in PIL directly
+
+                The pipeline accepts either a single image or a batch of images, which must then be passed as a string.
+                Images in a batch must all be in the same format: all as http links, all as local paths, or all as PIL
+                images.
+            threshold (:obj:`float`, `optional`, defaults to 0.9):
+                The probability necessary to make a prediction.
+
+        Return:
+            A list of dictionaries or a list of list of dictionaries containing the result. If the input is a single
+            image, will return a list of dictionaries, if the input is a list of several images, will return a list of
+            list of dictionaries corresponding to each image.
+
+            The dictionaries contain the following keys:
+
+            - **label** (:obj:`str`) -- The label identified by the model.
+            - **score** (:obj:`float`) -- The score attributed by the model for that label.
+            - **box** (:obj:`List[Dict[str, int]]`) -- The bounding box of detected object in image's original size.
+        """
+        is_batched = isinstance(images, list)
+
+        if not is_batched:
+            images = [images]
+
+        images = [self.load_image(image) for image in images]
+
+        with torch.no_grad():
+            inputs = self.feature_extractor(images=images, return_tensors="pt")
+            outputs = self.model(**inputs)
+
+            target_sizes = torch.IntTensor([[im.height, im.width] for im in images])
+            annotations = self.feature_extractor.post_process(outputs, target_sizes)
+            for i, annotation in enumerate(annotations):
+                keep = annotation["scores"] > threshold
+                scores = annotation["scores"][keep]
+                labels = annotation["labels"][keep]
+                boxes = annotation["boxes"][keep]
+
+                annotation["scores"] = scores.tolist()
+                annotation["labels"] = [self.model.config.id2label[label.item()] for label in labels]
+                annotation["boxes"] = [self._get_clockwise_vertices(box) for box in boxes]
```

Comment:
I am not a big fan of this format.

bounding boxes are quite standarly defined by either:

- x, y, width, height
- x_min, x_max, y_min, y_max

A list of (x, y) is both uncommon and too general (we can't be sure it's a rectangle from a user's perspective so drawing anything will require additional computing)

Reply:
Implementing `x_min, x_max, y_min, y_max` format in the update 👍